### PR TITLE
add method to get metric by name

### PR DESCRIPTION
**Pull request by ofweb**
[ch-53808]

### CODE REVIEW

**Review by chribsen — APPROVED** [review]

**Comment by chribsen** [comment]
don't you need a way to clear a specific metric immediately afterwards? Otherwise, you may end up reporting the same value again.

**Comment by ofweb** [comment]
> don't you need a way to clear a specific metric immediately afterwards? Otherwise, you may end up reporting the same value again.

Good point, I'll add that

File: src/metric.js [hunk 1]
```diff
@@ -158,6 +158,18 @@ class MetricRegistry {
...
+  getMetric(name) {
+    return this.metrics[name]
+  }
+
+  clearMetric(name) {
+    delete this.metrics[name]
+  }
+
+  getMetricNames() {
+    return Object.keys(this.metrics).sort((a, b) => a.localeCompare(b))
```
Comment (chribsen) [inline src/metric.js:171]: Why do you need it sorted?

Comment (ofweb) [reply src/metric.js]: good point, I moved that to the test

File: types/@connectedcars/logutil/index.d.ts [hunk 2]
```diff
@@ -5,10 +5,34 @@ export function warn(...args: any[]): void
...
+
+export interface GaugeMetric<K extends string> {
+  name: K,
+  type: 'GAUGE',
+  value: number,
+
+  labels: MetricLabels
+}
+export interface CumulativeMetric<K extends string> {
+  name: K,
```
Comment (chribsen) [inline types/@connectedcars/logutil/index.d.ts:18]: Why not just write string?

**Review by chribsen — APPROVED** [review]

**Pull request merged**